src/commands/clone.js

```diff
@@ -41,10 +42,36 @@ function clone (opts) {
   var onExit = require('../lib/exit')
   var parseArgs = require('../parse-args')
   var debug = require('debug')('dat')
-
   var parsed = parseArgs(opts)
+
   opts.key = parsed.key || opts._[0] // pass other links to resolver
   opts.dir = parsed.dir
+
+  // corrects parse-args.js default output for when no second argument for output directory is provided.
+  if (parsed.keyToDirSwitch) opts.dir = path.resolve('.')
+
+  // variables needed to check if opts.key is on system and leads to a dat.json file.
+  var datPath = opts.key
+  var onSystem = fs.existsSync(datPath)
+  var isDir = false
+  var isDatJson = false
+
+  // if the file or directory is on system check to see whether the path is to a dat.json file or directory.
+  if (onSystem) {
+    isDir = fs.lstatSync(datPath).isDirectory()
+    isDatJson = (fs.lstatSync(datPath).isFile() && (path.basename(datPath) === 'dat.json'))
+  }
+
+  // if file is a dat.json resolve absolute path. else if directory try to resolve path to dat.json.
+  if (isDatJson) {
+    datPath = path.resolve(datPath)
+  } else if (isDir) {
+    datPath = path.join(datPath + '/dat.json')
```

Comment:
`path.join(datPath, 'dat.json')` (the `/` will not give the expected result in windows, so `path.join` replaces it with whatever is correct for the OS).